examples/flutter_gallery/android/app/build.gradle

```diff
@@ -11,6 +11,14 @@ if (flutterRoot == null) {
     throw new GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
 }
 
+def flutterVersion = file("$flutterRoot/version").text
+def flutterVersionComponents = flutterVersion.split(/[^0-9]+/)
+// Let the integer version code of xx.yy.zz-pre.nnn be xxyyzznnn.
+int flutterVersionCode = Math.min(flutterVersionComponents[0].toInteger(), 99) * 10000000
+flutterVersionCode += Math.min(flutterVersionComponents[1].toInteger(), 99) * 100000
+flutterVersionCode += Math.min(flutterVersionComponents[2].toInteger(), 99) * 1000
+flutterVersionCode += Math.min(flutterVersionComponents[3]?.toInteger() ?: 0, 999)
```

Comment:
zz will almost certainly be greater than 99 at some point

Reply:
also, nnn will always be 0 in the cases where we actually publish the gallery, right? if we only publish for dev or beta builds. So maybe we can remove the nnn case, and spread those digits to z and y.

Reply:
I shuffled the digits around just a bit. Unfortunately the play store is a bit yolo about stuff like this and the int value can only go up from this point. 

I would also imagine that it should be possible to manually build and deploy from any point in time.